### PR TITLE
Mimir helm chart - Add graphite proxy support

### PR DESCRIPTION
**Pull request by jesusvazquez**
This PR is the new version of https://github.com/grafana/helm-charts/pull/872 now that mimir's helm chart has been migrated over to this repository. What this PR does:
- Extends mimir-distributed chart values.yml with an option to enable graphite proxy to run alongside mimir.
- If `graphite.enabled: true`:
  - [x] Added new  graphite-querier deployment
    - [x] + 2 memcached instances for the querier
  - [x] Added new graphite-write-proxy deployment
  - [x] Added new graphite-web deployment (This is an stock graphite instance that handles graphite function for which the proxy doesn't have support yet)
  - [x] Extends the gateway configuration to expose graphite api routes.
  - [x] Updates docs

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-web-dep.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+            - name: GRAPHITE_USE_WORKER_POOL
+              value: "false"
+            {{- range $key, $value := .Values.graphite_web.extraEnvVars }}
+            - name: {{ $key }}
+              value: "{{ $value }}"
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 80
+              protocol: TCP
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-web-dep.yaml:56]:
```suggestion
            - name: http-metrics
              containerPort: {{ include "mimir.serverHttpListenPort" . }}
              protocol: TCP
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-web-svc.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  annotations:
+    {{- toYaml .Values.graphite_web.service.annotations | nindent 4 }}
+  namespace: {{ .Release.Namespace | quote }}
+spec:
+  type: ClusterIP
+  ports:
+    - port: {{ include "mimir.serverHttpListenPort" . }}
+      protocol: TCP
+      name: graphite-http
+      targetPort: 80
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-web-svc.yaml:20]:
```suggestion
      name: http-metrics
      targetPort: http-metrics
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-svc.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+kind: Service
+metadata:
+  name: {{ include "mimir.resourceName" (dict "ctx" . "component" "graphite-querier") }}
+  labels:
+    {{- include "mimir.labels" (dict "ctx" . "component" "graphite-querier") | nindent 4 }}
+    {{- with .Values.graphite_querier.service.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- toYaml .Values.graphite_querier.annotations | nindent 4 }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-svc.yaml:12]:
```suggestion
    {{- toYaml .Values.graphite_querier.service.annotations | nindent 4 }}
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 4]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+    tag: 8-7ebde4c60
+    pullPolicy: IfNotPresent
+
+  service:
+    annotations: {}
+    labels: {}
+
+  deployment:
+    annotations: {}
+    labels: {}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2075]: For other deployments and statefulsets we only have e.g. `querier.annotations`; so without the `deployment` block. I prefer the separation you have here, but it's probably better if we are consistent.

I see `deployment.labels` isn't used in the template, so perhaps this is a WIP artefact? Other deployments/statefulsets don't have the `labels` configurable.

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+      {{- end }}
+      containers:
+        - name: graphite-write-proxy
+          image: "{{ include "mimir.imageReference" . }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          args:
+            - -target=graphite-write-proxy
+            - -config.file=/etc/mimir/mimir.yaml
+            - -license.path=/license/license.jwt
+            - -server.http-listen-port={{ include "mimir.serverHttpListenPort" . }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml:53]: You shoulnd't need to set this. It's already taken care of by the default `mimir.config`


```suggestion
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+            - -graphite.enabled=true
+            - -graphite.write-proxy.distributor-client.address=dns:///{{ template "mimir.fullname" . }}-distributor.{{ .Release.Namespace }}.svc:{{ include "mimir.serverGrpcListenPort" .  }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml:55]: Can you move these to the `mimir.config` yaml config?

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+            {{- toYaml .Values.graphite_write_proxy.readinessProbe | nindent 12 }}
+          resources:
+            {{- toYaml .Values.graphite_write_proxy.resources | nindent 12 }}
+          securityContext:
+            readOnlyRootFilesystem: true
+          env:
+            {{- with .Values.global.extraEnv }}
+              {{ toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with .Values.graphite_querier.env }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml:91]:
```suggestion
            {{- with .Values.graphite_write_proxy.env }}
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+              {{ toYaml . | nindent 12 }}
+            {{- end }}
+          envFrom:
+            {{- with .Values.global.extraEnvFrom }}
+              {{- toYaml . | nindent 12 }}
+            {{- end }}
+            {{- with .Values.admin_api.extraEnvFrom }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml:98]:
```suggestion
            {{- with .Values.graphite_write_proxy.extraEnvFrom }}
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+      namespace: {{ .Release.Namespace | quote }}
+    spec:
+      serviceAccountName: {{ template "mimir.serviceAccountName" . }}
+      {{- if .Values.graphite_write_proxy.priorityClassName }}
+      priorityClassName: {{ .Values.graphite_write_proxy.priorityClassName }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.graphite_write_proxy.securityContext | nindent 8 }}
+      initContainers:
+        {{- with .Values.graphite_querier.initContainers }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-dep.yaml:36]:
```suggestion
        {{- with .Values.graphite_write_proxy.initContainers }}
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 10]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+      maxUnavailable: 1
+
+  terminationGracePeriodSeconds: 180
+
+  env: []
+  extraEnvFrom: []
+  tolerations: []
+  podDisruptionBudget: {}
+  initContainers: []
+  extraContainers: []
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:1991]: This isn't being used in the deployment template. Can you add see `compactor-statefulset.yaml` as an example

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 11]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+  extraVolumes: []
+  extraVolumeMounts: []
+  extraPorts: []
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:1994]: This isn't being used in the deployment template. But I also can't find other components that use it. Can you remove it?
```suggestion
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-svc.yaml [hunk 12]
```diff
@@ -0,0 +1,27 @@
+{{- if .Values.graphite.enabled -}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "mimir.resourceName" (dict "ctx" . "component" "graphite-write-proxy") }}
+  labels:
+    {{- include "mimir.labels" (dict "ctx" . "component" "graphite-write-proxy") | nindent 4 }}
+    {{- with .Values.graphite_write_proxy.labels }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-svc.yaml:8]:
```suggestion
    {{- with .Values.graphite_write_proxy.service.labels }}
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-svc.yaml [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- toYaml .Values.graphite_write_proxy.annotations | nindent 4 }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy-svc.yaml:12]:
```suggestion
    {{- toYaml .Values.graphite_write_proxy.service.annotations | nindent 4 }}
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-dep.yaml [hunk 14]
```diff
@@ -0,0 +1,140 @@
...
+  selector:
+    matchLabels:
+      {{- include "mimir.selectorLabels" (dict "ctx" . "component" "graphite-querier") | nindent 6 }}
+  strategy:
+    {{- toYaml .Values.graphite_querier.strategy | nindent 4 }}
+  template:
+    metadata:
+      labels:
+        {{- include "mimir.podLabels" (dict "ctx" . "component" "graphite-querier") | nindent 8 }}
+        {{- with .Values.graphite_write_proxy.podLabels }}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-dep.yaml:22]:
```suggestion
        {{- with .Values.graphite_querier.podLabels }}
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 15]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+              matchExpressions:
+                - key: target
+                  operator: In
+                  values:
+                    - graphite-write-proxy
+            topologyKey: 'kubernetes.io/hostname'
+
+  annotations: {}
+  persistence:
+    subPath:
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2033]: Similar question as for the querier - does it need persistence? I don't see the `/data` dir being used anywhere in the config.

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 16]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+
+  terminationGracePeriodSeconds: 180
+
+  env: []
+  extraEnvFrom: []
+  tolerations: []
+  podDisruptionBudget: {}
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2054]: None of the querier, write proxy, web have a pod disruption budget. Only the 2 caches have it. Can you add files with the PDB for the components too?

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 10 above)
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:1991]: Same comment for `graphite_write_proxy`

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 17]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+  extraVolumes: []
+  extraVolumeMounts: []
+  extraPorts: []
+
+graphite_web:
+  replicas: 1
+
+  image:
+    repository: docker.io/raintank/graphite-mt
+    tag: 8-7ebde4c60
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2066]: This particular tag is 2 years-old. There is a v2 pushed. Should we use that instead?
```suggestion
    tag: 8-7ebde4c60-v2
```

**Review by dimitarvdimitrov — COMMENTED** [review]
Thanks for the contribution :) I left only a couple of important comments. Most are just copy-past mistakes.

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 15 above)
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/values.yaml:2032]: They do, otherwise the pods fail to start. Its requirement of GEM.
```
level=info ts=2022-08-19T11:19:55.923295616Z caller=main.go:231 msg="starting Grafana Enterprise Metrics" version="(version=r197-c28774dd, branch=weekly-r197, revision=c28774dd)"
level=error ts=2022-08-19T11:19:55.92344861Z caller=server_util.go:26 original_caller=main.go:235 msg="error running Grafana Enterprise Metrics" err="open /data/metrics-activity.log: read-only file system\nfailed to create activity file\ngithub.com/grafana/mimir/pkg/util/activitytracker.getMappedFile\n\t/backend-enterprise/vendor/github.com/grafana/mimir/pkg/util/activitytracker/activity_tracker.go:177\ngithub.com/grafana/mimir/pkg/util/activitytracker.NewActivityTracker\n\t/backend-enterprise/vendor/github.com/grafana/mimir/pkg/util/activitytracker/activity_tracker.go:60\ngithub.com/grafana/mimir/pkg/mimir.(*Mimir).initActivityTracker\n\t/backend-enterprise/vendor/github.com/grafana/mimir/pkg/mimir/modules.go:141\ngithub.com/grafana/dskit/modules.(*Manager).initModule\n\t/backend-enterprise/vendor/github.com/grafana/dskit/modules/modules.go:120\ngithub.com/grafana/dskit/modules.(*Manager).InitModuleServices\n\t/backend-enterprise/vendor/github.com/grafana/dskit/modules/modules.go:92\ngithub.com/grafana/mimir/pkg/mimir.(*Mimir).Run\n\t/backend-enterprise/vendor/github.com/grafana/mimir/pkg/mimir/mimir.go:551\ngithub.com/grafana/backend-enterprise/pkg/enterprise/mimir/init.(*MimirEnterprise).Run\n\t/backend-enterprise/pkg/enterprise/mimir/init/mimir.go:158\nmain.main\n\t/backend-enterprise/cmd/enterprise-metrics/main.go:232\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1571\nerror initialising module: activity-tracker\ngithub.com/grafana/dskit/modules.(*Manager).initModule\n\t/backend-enterprise/vendor/github.com/grafana/dskit/modules/modules.go:122\ngithub.com/grafana/dskit/modules.(*Manager).InitModuleServices\n\t/backend-enterprise/vendor/github.com/grafana/dskit/modules/modules.go:92\ngithub.com/grafana/mimir/pkg/mimir.(*Mimir).Run\n\t/backend-enterprise/vendor/github.com/grafana/mimir/pkg/mimir/mimir.go:551\ngithub.com/grafana/backend-enterprise/pkg/enterprise/mimir/init.(*MimirEnterprise).Run\n\t/backend-enterprise/pkg/enterprise/mimir/init/mimir.go:158\nmain.main\n\t/backend-enterprise/cmd/enterprise-metrics/main.go:232\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1571"
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 18]
```diff
@@ -1911,6 +1911,321 @@ gateway:
...
+            labelSelector:
+              matchExpressions:
+                - key: target
+                  operator: In
+                  values:
+                    - graphite-querier
+            topologyKey: 'kubernetes.io/hostname'
+
+  annotations: {}
+  persistence:
```
[the remark this answers is not in the document]
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/values.yaml:1979]: The mimir image does otherwise it fails to start. The persistence.subPath key here is left just in case the user wants to change the volume path I guess?

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml [hunk 19]
```diff
@@ -0,0 +1,22 @@
...
+metadata:
+  name: {{ template "mimir.fullname" . }}-graphite-schemas
+  labels:
+    app: {{ template "mimir.name" . }}
+    chart: {{ template "mimir.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+  namespace: {{ .Release.Namespace | quote }}
+data:
+{{- range  $storageSchemas := toStrings .Values.graphite_querier.schemasConfiguration.storageSchemas }}
```
[the remark this answers is not in the document]
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml:11]: I'm getting an error when I attempt this: 
```
Error: UPGRADE FAILED: YAML parse error on mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml: error converting YAML to JSON: yaml: line 16: could not find expected ':'
```
What I wrote: 
```
data:
  storage-schemas.conf: {{ .Values.graphite_querier.schemasConfiguration.storageSchemas }}
  storage-aggregations.conf: {{ .Values.graphite_querier.schemasConfiguration.storageAggregations }}
```

This is what its producing: 
```
# Source: mimir-distributed/templates/graphite-proxy/graphite-queri
er-schemas-configmap.yaml
apiVersion: v1
kind: ConfigMap
metadata:
  name: graphite-demo-mimir-graphite-schemas
  labels:
    helm.sh/chart: mimir-distributed-3.1.0-weekly.197
    app.kubernetes.io/name: mimir
    app.kubernetes.io/instance: graphite-demo
    app.kubernetes.io/component: graphite-schemas
    app.kubernetes.io/version: "r197"
    app.kubernetes.io/managed-by: Helm
  namespace: "default"
data:
  storage-schemas.conf: [default]
pattern = .*
intervals = 0:1s
retentions = 10s:8d,10min:1y
  storage-aggregations.conf: [default]
aggregationMethod = avg
pattern = .*
xFilesFactor = 0.1
---
```

I think that because its a multiline string that is unquoted it fails to parse. This is why i was using range + toStrings.

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 18 above)
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:1979]: what does it fail with?

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml (hunk 19 above)
[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml:11]: try using 
```
data:
  storage-schemas.conf: {{ .Values.graphite_querier.schemasConfiguration.storageSchemas | indent 2 }}
  storage-aggregations.conf: {{ .Values.graphite_querier.schemasConfiguration.storageAggregations | indent 2 }}
```

helm does not understand that the surrounding file needs to be valid yaml, so it just literally inlines a block of text. you need to tell it to indent it by 2 so it doesnt break the yaml structure

[the remark this answers is not in the document]
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-schemas-configmap.yaml:11]: Thanks, make it work with this:
```
  storage-schemas.conf: |
{{ .Values.graphite_querier.schemasConfiguration.storageSchemas | indent 4 }}
  storage-aggregations.conf: |
{{ .Values.graphite_querier.schemasConfiguration.storageAggregations | indent 4 }}
```

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 18 above)
[the remark this answers is not in the document]
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/values.yaml:1979]: With this https://github.com/grafana/mimir/pull/2711#discussion_r950084807

[the remark this answers is not in the document]
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:1985]: replied in that comment

File: operations/helm/charts/mimir-distributed/values.yaml (hunk 15 above)
Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:2063]: ok nevermind, I see it's using an empty dir anyways. you might as well keep this.

File: operations/helm/charts/mimir-distributed/templates/_helpers.tpl [hunk 20]
```diff
@@ -338,6 +338,11 @@ Examples:
   "compactor" "compactor"
   "distributor" "distributor"
   "gateway" "gateway"
+  "graphite-querier" "graphite.querier"
+  "graphite-write-proxy" "graphite.write_proxy"
+  "graphite-web" "graphite.graphite_web"
+  "graphite-aggregation-cache" "graphite-aggregation-cache"
+  "graphite-metric-name-cache" "graphite-metric-name-cache"
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/_helpers.tpl:345]: can you sort these alphabetically? 

```suggestion
  "graphite-aggregation-cache" "graphite-aggregation-cache"
  "graphite-metric-name-cache" "graphite-metric-name-cache"
  "graphite-querier" "graphite.querier"
  "graphite-web" "graphite.graphite_web"
  "graphite-write-proxy" "graphite.write_proxy"
```

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml [hunk 21]
```diff
@@ -53,6 +53,7 @@ spec:
             - -target=graphite-write-proxy
             - -config.file=/etc/mimir/mimir.yaml
             - -license.path=/license/license.jwt
+            - -instrumentation.enabled=false
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml:56]: Out of curiosity - why is this needed?

**Review by dimitarvdimitrov — COMMENTED** [review]
left some minor comments, but I think this is mostly there :)

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml (hunk 21 above)
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml:56]: To avoid a common error in metrics enterprise image i.e
```level=warn ts=2022-08-24T06:54:40.097647Z caller=gatherer.go:71 msg="Unable to send system metrics" err="unable to push to distributor: rpc error: code = Unavailable desc = last resolver error: produced zero addresses"````

The instrumentation is always enabled as it thinks its always going to run some mimir component and expose cortex metrics but it is not the case when running graphite targets.

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-web/graphite-web-dep.yaml [hunk 22]
```diff
@@ -0,0 +1,65 @@
+{{- if and .Values.enterprise.enabled .Values.graphite.enabled -}}
```
[the remark this answers is not in the document]
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-web/graphite-web-dep.yaml:1]: Done :+1:

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml (hunk 21 above)
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-write-proxy/graphite-write-proxy-dep.yaml]: Just reenabled instrumentation

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-dep.yaml [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+            - -graphite.querier.metric-name-cache.memcached.addresses=dnssrvnoa+{{ template "mimir.fullname" . }}-graphite-metric-name-cache.{{ .Release.Namespace}}.svc:11211
+            - -graphite.querier.metric-name-cache.memcached.timeout=1s
+            {{- if .Values.minio.enabled }}
+            - -admin.client.backend-type=s3
+            - -admin.client.s3.endpoint={{ .Release.Name }}-minio.{{ .Release.Namespace }}.svc:9000
+            - -admin.client.s3.bucket-name=enterprise-metrics-admin
+            - -admin.client.s3.access-key-id=enterprise-metrics
+            - -admin.client.s3.secret-access-key=supersecret
+            - -admin.client.s3.insecure=true
+            {{- end }}
```
[the remark this answers is not in the document]
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier-dep.yaml]: Done

File: operations/helm/charts/mimir-distributed/templates/_helpers.tpl (hunk 20 above)
Comment (jesusvazquez) [reply operations/helm/charts/mimir-distributed/templates/_helpers.tpl]: Done :+1:

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 24]
```diff
@@ -2053,6 +2086,326 @@ gateway:
...
+
+    nodeSelector: {}
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: target
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2136]: target is an outdated label, so this affinity rule doesn't match anything you can use app.kubernetes.io/component

```suggestion
                  - key: app.kubernetes.io/component
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 25]
```diff
@@ -2053,6 +2086,326 @@ gateway:
...
+    nodeSelector: {}
+    # -- Affinity for graphite-write-proxy pods
+    affinity:
+      podAntiAffinity:
+        preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                  - key: target
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2207]: target is an outdated label, so this affinity rule doesn't match anything you can use app.kubernetes.io/component

```suggestion
                  - key: app.kubernetes.io/component
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 26]
```diff
@@ -2053,6 +2086,326 @@ gateway:
...
+    podAnnotations: {}
+    # -- Node selector for graphite-web pods
+    nodeSelector: {}
+    # -- Affinity for graphite-web pods
+    affinity:
+      podAntiAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+                - key: target
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2275]: target is an outdated label, so this affinity rule doesn't match anything you can use app.kubernetes.io/component

```suggestion
                  - key: app.kubernetes.io/component
```

**Review by dimitarvdimitrov — COMMENTED** [review]
I had missed some details around affinity rules

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, minus the docs link for the caches we spoke in about in a private convo

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 27]
```diff
@@ -2053,6 +2086,330 @@ gateway:
         hosts:
           - gateway.gem.example.com
 
+graphite:
+  # -- If true, enables graphite querier and graphite write proxy functionality.
+  # Read more in https://grafana.com/docs/metrics-enterprise/latest/graphite/
```
Comment (osg-grafana) [inline operations/helm/charts/mimir-distributed/values.yaml:2091]: @dimitarvdimitrov can you change this URL `.../enterprise-metrics/...`?

Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml:2091]: sure thing https://github.com/grafana/mimir/pull/2830
